crates/bevy_ecs/src/schedule/executor/multi_threaded.rs

```diff
@@ -309,6 +310,9 @@ impl MultiThreadedExecutor {
                 break;
             }
 
+            // TODO: do this earlier?
```

Comment:
Yes, you want to replace the `cell: &'scope SyncUnsafeCell<World>` with `cell: UnsafeWorldCell<'scope>` I believe.